build-tools-internal/src/integTest/groovy/org/elasticsearch/gradle/internal/precommit/transport/TransportTestExistPluginFuncTest.groovy

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.gradle.internal.precommit.transport
+
+import spock.lang.Shared
+
+import org.elasticsearch.gradle.fixtures.AbstractGradleFuncTest
+import org.elasticsearch.gradle.fixtures.AbstractGradleInternalPluginFuncTest
+import org.elasticsearch.gradle.fixtures.LocalRepositoryFixture
+import org.elasticsearch.gradle.internal.conventions.precommit.PrecommitPlugin
+import org.elasticsearch.gradle.internal.precommit.CheckstylePrecommitPlugin
+import org.gradle.api.Plugin
+import org.gradle.testkit.runner.TaskOutcome
+import org.junit.ClassRule
+
+class TransportTestExistPluginFuncTest extends AbstractGradleInternalPluginFuncTest {
+    Class<? extends TransportTestExistPrecommitPlugin> pluginClassUnderTest = TransportTestExistPrecommitPlugin.class
+
+    @Shared
+    @ClassRule
+    public LocalRepositoryFixture repository = new LocalRepositoryFixture()
+
+    def setup() {
+        configurationCacheCompatible = false
```

Comment:
why do we need to disable this for that plugin

Reply:
This PR breaks `BuildPluginFuncTest` it seems

Reply:
agree we probably don't need to disable `        configurationCacheCompatible = false` (it was failing for me before, but seems fine now)

re `BuildPluginFuncTest` failures. This is because that test will also trigger the TransportTestExistPlugin and does not have a `:server` project setup. I can disable it as it is done for thridPartyAudit already. wdyt?
```
   // tested elsewhere
            tasks.named('thirdPartyAudit').configure { enabled = false }
            tasks.named('transportTestExistCheck').configure { enabled = false }
            """
```